tests/test_derivative.py

```diff
@@ -0,0 +1,229 @@
+        y_adj_np = Fop.H @ x_global
+        assert_allclose(y, y_np, rtol=1e-14)
+        assert_allclose(y_adj, y_adj_np, rtol=1e-14)
+
+
+@pytest.mark.mpi(min_size=2)
+@pytest.mark.parametrize("par", [(par1), (par1b), (par1j), (par1e), (par2), (par2b),
+                                 (par2j), (par2e), (par3), (par3b), (par3j), (par3e),
+                                 (par4), (par4b), (par4j), (par4e)])
+def test_first_derivative_backward(par):
+    Fop_MPI = pylops_mpi.MPIFirstDerivative(dims=par['nz'], sampling=par['dz'],
+                                            kind="backward", edge=par['edge'],
+                                            dtype=par['dtype'])
+    x = pylops_mpi.DistributedArray(global_shape=par['nz'], dtype=par['dtype'],
+                                    partition=par['partition'])
+    x[:] = np.random.normal(rank, 10, x.local_shape)
+    x_global = x.asarray()
+    # Forward
+    y_dist = Fop_MPI @ x
+    y = y_dist.asarray()
+    # Adjoint
+    y_adj_dist = Fop_MPI.H @ x
+    y_adj = y_adj_dist.asarray()
+
+    if rank == 0:
+        Fop = pylops.FirstDerivative(dims=par['nz'], sampling=par['dz'],
+                                     kind="backward", edge=par['edge'],
+                                     dtype=par['dtype'])
+        assert Fop_MPI.shape == Fop.shape
+        y_np = Fop @ x_global
+        y_adj_np = Fop.H @ x_global
+        assert_allclose(y, y_np, rtol=1e-14)
+        assert_allclose(y_adj, y_adj_np, rtol=1e-14)
+
+
+@pytest.mark.mpi(min_size=2)
+@pytest.mark.parametrize("par", [(par1), (par1b), (par1j), (par1e), (par2), (par2b),
+                                 (par2j), (par2e), (par3), (par3b), (par3j), (par3e),
+                                 (par4), (par4b), (par4j), (par4e)])
+def test_first_derivative_centered(par):
+    for order in [3, 5]:
+        Fop_MPI = pylops_mpi.MPIFirstDerivative(dims=par['nz'], sampling=par['dz'],
+                                                kind="centered", edge=par['edge'],
+                                                order=order, dtype=par['dtype'])
+        x = pylops_mpi.DistributedArray(global_shape=par['nz'], dtype=par['dtype'],
+                                        partition=par['partition'])
+        x[:] = np.random.normal(rank, 10, x.local_shape)
+        x_global = x.asarray()
+        # Forward
+        y_dist = Fop_MPI @ x
+        y = y_dist.asarray()
+        # Adjoint
+        y_adj_dist = Fop_MPI.H @ x
+        y_adj = y_adj_dist.asarray()
+        if rank == 0:
+            Fop = pylops.FirstDerivative(dims=par['nz'], sampling=par['dz'],
+                                         kind="centered", edge=par['edge'],
+                                         order=order, dtype=par['dtype'])
+            assert Fop_MPI.shape == Fop.shape
+            y_np = Fop @ x_global
+            y_adj_np = Fop.H @ x_global
+            assert_allclose(y, y_np, rtol=1e-14)
+            assert_allclose(y_adj, y_adj_np, rtol=1e-14)
```

Comment:
We may want to also add a test (just for one of the kind) where the input array is Nd (say 2d)... I think, this is where the problem I mentioned in the general comment will show up if you were to have x as a DistributedArray with axis!=0... tests are a useful way to catch issues, not only to check correctness.

Of course, the test to add will be with axis=0, but I guess you get my point ;)